docs/modules/release-notes/pages/5-4-0.adoc

```diff
@@ -0,0 +1,353 @@
+= 5.4.0 Release Notes
+
+CAUTION: Starting with this release of Hazelcast Platform, the minimum supported Java version is 17.
+
+== New Features
+
+* [.enterprise]*Enterprise* **User Code Namespaces**: Enable deployment and redeployment of your custom classes. See the xref:clusters:user-code-namespaces.adoc[User Code Namespaces documentation].
+* [.enterprise]*Enterprise* **Thread-Per-Core (TPC)**: TPC is now generally available. You can enable this feature on the clients and cluster members for improved performance. See the xref:cluster-performance:thread-per-core-tpc.adoc[Thread-Per-Core (TPC) documentation].
+* [.enterprise]*Enterprise* **CPMap**:  Added CPMap as a minimal key-value CP data structure. See xref:data-structures:cpmap.adoc[CPMap documentation].
+https://github.com/hazelcast/hazelcast/pull/25802[#25802]
+
+== Breaking Changes
+
+* The `MergingValue` interface within the SPI package now requires the `getDeserializedValue()` method to be defined within implementations, replacing the `getValue()` definition.
+https://github.com/hazelcast/hazelcast/pull/25942[#25942]
+* Moved the MongoDB connector to the extensions module, that is, its classes and related dependencies relocated;
+if you are using Maven to install the connector, you must add `<classifier>jar-with-dependencies</classifier>` to your `pom.xml`.
+Also removed the permissions for MongoDB connector.
+https://github.com/hazelcast/hazelcast/pull/25744[#25744], https://github.com/hazelcast/hazelcast/pull/25701[#25701]
+* Changed the return values for `getMethodName()` of the following message task classes:
+** `MapFetchKeysMessageTask`: `iterator` -> `SecurityInterceptorConstants.ITERATOR_FETCH_KEYS`
+** `MapFetchWithQueryMessageTask.`: `iterator` -> `SecurityInterceptorConstants.ITERATOR_FETCH_WITH_QUERY`
+** `MapProjectionWithPredicateMessageTask`: `projectWithPredicate` -> `SecurityInterceptorConstants.PROJECT`
+** `ScheduledExecutorGetAllScheduledMessageTask`: `getAllScheduled` -> `SecurityInterceptorConstants.GET_ALL_SCHEDULED_FUTURES`
+** `ScheduledExecutorSubmitToPartitionMessageTask`: `submitToPartition` -> `SecurityInterceptorConstants.SCHEDULE_ON_PARTITION`
+** `ScheduledExecutorSubmitToTargetMessageTask`: `submitToAddress` -> `SecurityInterceptorConstants.SCHEDULE_ON_MEMBER`
+** `ScheduledExecutorTaskGetResultFromPartitionMessageTask`: `getResultTimeout` -> `SecurityInterceptorConstants.GET_RESULT`
+** `ScheduledExecutorTaskGetResultFromTargetMessageTask`: `getResultTimeout` -> `SecurityInterceptorConstants.GET_RESULT`
+** `SetGetAllMessageTask`: `iterator` -> `SecurityInterceptorConstants.GET_ALL`
```

Comment:
This change was subsequently reverted in https://github.com/hazelcast/hazelcast/pull/25551. Can you please check and amend the list?